google-cloud-datastore/lib/google/cloud/datastore/dataset/query_results.rb

```diff
@@ -85,18 +85,27 @@ class QueryResults < DelegateClass(::Array)
           # Datastore.
           attr_reader :batch_read_time
 
+          # Query explain metrics. This is only present when the
+          # [RunQueryRequest.explain_options][google.datastore.v1.RunQueryRequest.explain_options]
+          # is provided, and it is sent only once with the last response in the stream.
```

Comment:
Can you say explicitly something like "To retrieve metrics, iterate over all pages of results. The explain_metrics field will be set on the results object representing the final page" so users know exactly what they have to do to get the metrics.